File: 华为od2026题库/2026年4月15日/P14205/data/solution.cc
#include <vector>
#include <string>
#include <unordered_map>
#include <algorithm>
using namespace std;

class Solution {
public:
    vector<vector<int>> countKeys(string s) {
        unordered_map<char, int> cnt;
        int i = 0, n = s.size();

        while (i < n) {
            if (i + 1 < n && s[i] == 'u' && s[i + 1] == 'u') {
                cnt['j']++;
                i += 2;
            } else if (i + 1 < n && s[i] == 't' && s[i + 1] == 't') {
                cnt['b']++;
                i += 2;
            } else {
                cnt[s[i]]++;
                i++;
            }
        }

        vector<pair<char, int>> arr;
        for (auto &p : cnt) arr.push_back(p);

        sort(arr.begin(), arr.end(), [](const pair<char, int>& a, const pair<char, int>& b) {
            if (a.second != b.second) return a.second > b.second;
            return a.first < b.first;
        });

        vector<vector<int>> ans;
        for (auto &p : arr) {
            ans.push_back({encode(p.first), p.second});
        }
        return ans;
    }

private:
    int encode(char c) {
        if (c >= '0' && c <= '9') return c - '0';
        return c - 'a' + 10;
    }
};

File: 华为od2026题库/2026年4月15日/P14205/data/solution.java
import java.util.*;

class Solution {
    public List<List<Integer>> countKeys(String s) {
        Map<Character, Integer> cnt = new HashMap<>();
        int i = 0, n = s.length();

        while (i < n) {
            if (i + 1 < n && s.charAt(i) == 'u' && s.charAt(i + 1) == 'u') {
                cnt.put('j', cnt.getOrDefault('j', 0) + 1);
                i += 2;
            } else if (i + 1 < n && s.charAt(i) == 't' && s.charAt(i + 1) == 't') {
                cnt.put('b', cnt.getOrDefault('b', 0) + 1);
                i += 2;
            } else {
                char c = s.charAt(i);
                cnt.put(c, cnt.getOrDefault(c, 0) + 1);
                i++;
            }
        }

        List<Character> keys = new ArrayList<>(cnt.keySet());
        Collections.sort(keys, (a, b) -> {
            int ca = cnt.get(a), cb = cnt.get(b);
            if (ca != cb) return cb - ca;
            return a - b;
        });

        List<List<Integer>> ans = new ArrayList<>();
        for (char c : keys) {
            List<Integer> pair = new ArrayList<>();
            pair.add(encode(c));
            pair.add(cnt.get(c));
            ans.add(pair);
        }
        return ans;
    }

    private int encode(char c) {
        if (c >= '0' && c <= '9') return c - '0';
        return c - 'a' + 10;
    }
}
